peps/pep-0705.rst

```diff
@@ -185,111 +381,180 @@ Discussion:
     def f(a: A) -> None:
         print(a['x'] or 0)
 
-    b: B = {'x': 0}
+    b: B = {'x': 1}
     f(b)  # Accepted by type checker
 
-* A ``TypedDict`` or ``TypedMapping`` type with a required key is consistent with a ``TypedMapping`` type where the same key is a non-required key, again unlike relationships between two ``TypedDict`` types. Example::
+* An TypedDict type ``A`` with no explicit key ``'x'`` that allows other keys is not consistent with a TypedDict type with a non-required key ``'x'``, since at runtime the key ``'x'`` could be present and have an incompatible type (which may not be visible through ``A`` due to structural subtyping). The only exception to this rule is if ``'x'`` is non-required, read-only and of type ``Any``.
```

Comment:
```suggestion
* A TypedDict type ``A`` with no explicit key ``'x'`` that allows other keys is not consistent with a TypedDict type with a non-required key ``'x'``, since at runtime the key ``'x'`` could be present and have an incompatible type (which may not be visible through ``A`` due to structural subtyping). The only exception to this rule is if ``'x'`` is non-required, read-only and of type ``Any``.
```

Reply:
Separately, what if x is of type `x: NotRequired[ReadOnly[object]]`? Seems like then it would also be safe.

Reply:
I updated this by the way. Hadn't realised `object` was also a top type.